controller/src/main/java/org/jboss/as/controller/persistence/yaml/YamlConfigurationExtension.java

```diff
@@ -176,10 +193,28 @@ public void processOperations(ImmutableManagementResourceRegistration rootRegist
         for (Map.Entry<String, Object> deployment : deployments.entrySet()) {
             processUnmanagedDeployments(rootRegistration, deployment, xmlOperations, postExtensionOps);
         }
+        List<ParsedBootOp> reorderedList = new ArrayList<>(postExtensionOps.size());
+        for (ParsedBootOp op : postExtensionOps) {
+            if (parallelBootOp != null && isSubsystemOperation(op)) {
+                //The new operations created from the YAML are added to the parallel boot operation enclosing all the subsystem operations
```

Comment:
Perhaps add a comment line before this:

// ModelControllerImpl doesn't include subsystem ops directly in the postExtensionsOps param.
// So, if postExtensionOps contains a subsystem op, it's because we've created it from YAML.